server/opts.go

```diff
@@ -417,8 +534,15 @@ func parseCluster(cm map[string]interface{}, opts *Options) error {
 				opts.Routes = append(opts.Routes, url)
 			}
 		case "tls":
-			tlsm := mv.(map[string]interface{})
-			tc, err := parseTLS(tlsm)
+			var (
+				tc  *TLSConfigOpts
+				err error
+			)
+			if pedantic {
+				tc, err = parseTLS(tk, opts)
+			} else {
+				tc, err = parseTLS(mv, opts)
```

Comment:
This checks are a bit repetitive and would likely be difficult to maintain... I'm trying in another branch to refactor this and remove them using less interface and changing instead the `parseXYZ` methods to use the `token` type instead:

```diff
- func parseTLS(v interface{}, opts *Options) (*TLSConfigOpts, error) {
+ func parseTLS(i token, opts *Options) (*TLSConfigOpts, error) {
```

 https://github.com/wallyqs/gnatsd/commit/f379707210bf5742baa41facc2aa3a0fc91be8dd#diff-7b256a4f472c7d4b9e1931f2f1073b66L383

Reply:
Yes, looks better. Do you think we should hold on with current PR and wait to see if that branch could replace it?

Reply:
I think that branch would need some more changes so not yet ready, thinking maybe would be better to expose the token at the parser level to avoid calling `unwrap` to get the values from all the `map[string]interface{}` each time `range` is used (change them to `map[string]token` maybe but that would be a breaking change).

One thing with that branch is that is that it relies on always running the parser in pedantic mode and uses `ParseFileWithChecks` to use the token type (then the users of the parse decides whether to report or not only when `-t` is specified).

```go
	// Pedantic reporting is optional but we always parse
	// including the context from the tokens.
	m, err = conf.ParseFileWithChecks(configFile)
```

Right now unless `-t` is specified, `ParseFile` is still used so parser works as it used too.